script.js

```diff
@@ -149,9 +149,63 @@ function addCheckboxes() {
 		checkbox = document.createElement('input');
 		checkbox.type = 'checkbox';
 		checkbox.value = sha;
+		checkbox.onclick = highlightRange;
 		checkbox.className = 'commit-compare-checkbox';
 		
 		listItem.appendChild(checkbox);
+
+		function highlightRange() {
+			var allCheckboxes = getAllCheckboxElements(),
+				selectedCheckboxes = getSelectedCheckboxes(allCheckboxes),
+				foundFirstSelectedCheckbox = false,
+				foundSecondSelectedCheckbox = false;
+				
+			if(selectedCheckboxes.length < 2) {
+				enableAllCheckboxes();
+				return;
+			}
+			
+			allCheckboxes.forEach(disableBasedOnSelection);
+			
+			function disableBasedOnSelection(checkbox) {
+				if(!foundFirstSelectedCheckbox) {
+					disableUntilFirstCheckboxFound(checkbox);
+				} 
+				
+				if(!foundSecondSelectedCheckbox) {
+					verifyIsSecondSelected(checkbox);
+				} else {
+					disableAndUncheckAfterSecondCheckboxFound(checkbox);
+				}
+			}
+			
+			function disableAndUncheckAfterSecondCheckboxFound(checkbox) {
+				checkbox.disabled = true;
+				checkbox.checked = false;
+			}
+			
+			function disableUntilFirstCheckboxFound(checkbox) {
+				if(checkbox.value !== selectedCheckboxes[0].value) {
+					checkbox.disabled = true;
+				} else {
+					foundFirstSelectedCheckbox = true;
+				}
+			}
```

Comment:
A small performance improvement would be to stop the loop when we have found the `firstSelectedCheckbox`. forEach can however unfortunately not be stopped in a clean way and changing it to a normal for loop makes it less readable in my opinion.